### PR TITLE
[POL-208] Update react-scripts to v5

### PR DESCRIPTION
**Pull request by zarsky-broad**
Working, but beware that this changes the console output when developing locally. The webpack bundle output is shown now.

### CODE REVIEW

**Review by cahrens — APPROVED** [review]
This is great and should help reduce the dependabot noise! Please create/link a ticket for compliance.

I checked out the branch, did `yarn install`, and was able to run with no issues. The console info is a bit noisier, but doesn't particularly bother me. I did some spot-checking in Terra, but would also assume that integration tests would bomb if there was an issue.

File: config-overrides.js [hunk 1]
```diff
@@ -30,7 +30,8 @@ module.exports = {
         enabled: env === 'production',
         hashEnabled: { 'style-src': false },
         nonceEnabled: { 'style-src': false }
-      }))
+      })),
+      _.update('ignoreWarnings', (old = []) => _.concat(old, /Failed to parse source map/))
```
Comment (kyuksel) [inline config-overrides.js:34]: Why was this necessary? Asking for my edification.

**Review by kyuksel — APPROVED** [review]
Thank you for the upgrade! LGTM. Did some spot checks on the UI. 

When merging, let's send out a message on `#terra-ui` to give a heads-up about the console display changes.

File: config-overrides.js (hunk 1 above)
Comment (zarsky-broad) [reply config-overrides.js:34]: Try removing it and running locally and watch the output

**Comment by zarsky-broad** [comment]
I just added a few things to the `.yarnrc`; these make implicitly defined dependencies explicit. It gets rid of some yarn warnings when installing, and also makes us [mostly] PnP-ready.

**Comment by cahrens** [comment]
> I just added a few things to the `.yarnrc`; these make implicitly defined dependencies explicit. It gets rid of some yarn warnings when installing, and also makes us [mostly] PnP-ready.

Thanks, @zarsky-broad. What does "PnP-ready" mean?

**Comment by zarsky-broad** [comment]
@cahrens check it out: https://yarnpkg.com/features/pnp

**Comment by cahrens** [comment]
> @cahrens check it out: https://yarnpkg.com/features/pnp

@zarsky-broad Thanks! What are your thoughts about when (and how) we should try going to PnP?

**Comment by zarsky-broad** [comment]
I'm working on a PR for it now.
